RIOT-OS/RIOT
# boards/mips-malta: remove board.

## Contribution description

The mips-malta board is a maintainance burden, has no working UART input and is unobtainable and thus must be removed.

Given that most likely nobody runs this board (see testing procedure) and it has been broken for several releases, I suggest removing it right away without deprecation warning. The failures/expect errors should serve as warnings anyways.

### 1. Unobtainable board

The mips-malta board is not an off-the-shelf part. A quick web search only show the MIPS website where one is told to "contact sales".

I could find it on ebay, used, at €155 and from single seller.

Not having access to the board means:

a. We cannot maintain it. In fact it could be broken right now.
b. Potential RIOT uses have not access to the board either. In other words, it is pointless to run on hardware nobody has.

###  2. No working UART input

Not all applications need UART input, but that is no excuse for not supporting it:

a. Makes development & debugging way harder.
b. It is impossible to run interactive tests.
 b.1. Constrains the rest of the platforms by providing an incentive to not make tests interactive.
c. The lack of UART is a witness to the poor quality of the port.

I want to stress point (c). If something as basic as a serial port cannot work, how can we expect more complex functionality to work. The answer is impossible to know, because of point (1).

### 3. Maintainance burden

The RIOT project has limited time and human resources which can be better spent.

a. Compiling for mips-malta wastes CPU time.
b. Blacklisting the board in the test wastes contributor's time.
c. Adapting the board's makefile during build system rework takes time and makes the reworks harder.
 c.1. Add to that that the changes are most of the time not even tested on the board because of (1). Look at the github issues/PRs and you will see it.
d. Developers usually stick to the lowest common denominator. Issue (2) sets this denominator unacceptably low.

### MIPS platform in general

In commits I will address general issues in the MIPS platform and why it should all be removed.




## Testing procedure

All the other boards should continue to work without problems. 

In master you can verify that not even the minimal test works:

```· BOARD=mips-malta make -C tests/minimal
make: Entering directory '/home/jcarrano/source/masterRIOT/tests/minimal'
There are unsatisfied feature requirements: periph_uart


EXPECT ERRORS!
```

## Issues/PRs references

(Please help me complete this list)
MIPS Issues: #11788
